include/OpenSpaceToolkit/Physics/Manager.hpp

```diff
@@ -0,0 +1,148 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Physics_Manager__
+#define __OpenSpaceToolkit_Physics_Manager__
+
+#include <filesystem>
+#include <mutex>
+
+#include <OpenSpaceToolkit/Core/FileSystem/Directory.hpp>
+#include <OpenSpaceToolkit/Core/Type/String.hpp>
+
+#include <OpenSpaceToolkit/Physics/Time/Duration.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+
+namespace ostk
+{
+namespace physics
+{
+
+using ostk::core::filesystem::Directory;
+using ostk::core::filesystem::File;
+using ostk::core::filesystem::Path;
+using ostk::core::type::String;
+
+using ostk::physics::time::Duration;
+using ostk::physics::time::Instant;
+
+/// @brief                      OSTk manager base class (thread-safe)
+///
+///                             The base manager defines methods for tracking and checking the manifest file.
+
+class Manager
+{
+   public:
+    enum class Mode
+    {
+
+        Manual,    ///< Manually load and unload bulletins
+        Automatic  ///< Automatically fetch, load and unload bulletins (from remote repositories)
+
+    };
+
+    /// @brief                  Copy constructor (deleted)
+
+    Manager(const Manager& aManager) = delete;
+
+    /// @brief                  Copy assignment operator (deleted)
+
+    Manager& operator=(const Manager& aManager) = delete;
+
+    /// @brief                  move constructor (deleted)
+
+    Manager(Manager&&) = delete;
+
+    /// @brief                  move assignment operator (deleted)
+
+    Manager& operator=(Manager&&) = delete;
+
+    /// @brief                  Destructor
+
+    virtual ~Manager() = default;
+
+    /// @brief              Get manager mode
+    ///
+    /// @return             Manager mode
+
+    Manager::Mode getMode() const;
+
+    /// @brief              Get local repository
+    ///
+    /// @return             Local repository
+
+    Directory getLocalRepository() const;
+
+    /// @brief              Get local repository timeout
+    ///
+    /// @return             Local repository timeout
```

Comment:
Can you add a small description on what the lock timeout is for? Either here and/or in the README?